### PR TITLE
Two belling fixes

### LINKED ISSUE #12258: Pressing Media keys in terminal after changing bell sound replays the bell
Regressed in #11511

#### Repro steps
1. change the bell sound to honk, 
2. Emit a bell (tab in `bash` with an empty cmdline is easy)
3. press the media Play button with the Terminal focused

#### Expected
Media toggles play/pause

#### Actual
get honk

### LINKED ISSUE #12276: Can't override bell sound?
1. put the honks in `profiles.defaults`
2. set a profile to `bellSound: null`
3. bell Terminal

expected:

No honks, normal bell

actual:

N O T H I N G

### PR DESCRIPTION
**Pull request by zadjii-msft**
Sorry for combining two fixes in one PR. I can separate if need be.

* [x] Closes #12276:
  - `"bellSound": null` didn't work. This one was easier, and is atomically in bcc2ca04fc14f39f37849b4bd837ad6cdb4cdaaa. Basically, we would deserialize that as an array with a single empty string in it, which we'd try to then play. I think it's more idomatic to have that deserialized as an empty array, which correctly falls back to playing the default sound.
* [x] Closes #12258: 
  - This one is the majority of the rest of the PR. If you leave the MediaPlayer open, then the media keys will _affect the Terminal_. More importantly, once the bell sounds, they'd replay the bell, which is insane. So the fix is to re-create the media player when we need it. We do this per-pane for simpler lifetime tracking. I'm not worried about the overhead of creating a mediaplayer here, since we're already throttling bells.
* Originally added in #11511
* [x] Tested manually
  - Use [`no.mp4`](https://www.youtube.com/watch?v=x2w9TyCv2gk) for this since that's like, 17s long
  - Checked that closing panes / the terminal while a bell was playing didn't crash
  - Playing a bunch of bells at once works
  - closing a pane stops the bell it's playing
  - once the bell stops, the media keys went back to working for Spotify
* [x] I work here

### CODE REVIEW

File: src/cascadia/TerminalApp/Pane.cpp [hunk 1]
```diff
@@ -1061,17 +1030,48 @@ void Pane::_ControlConnectionStateChangedHandler(const winrt::Windows::Foundatio
...
     co_await winrt::resume_foreground(_root.Dispatcher());
-    if (auto pane{ weakThis.get() })
+    if (auto pane{ weakThis.lock() })
     {
-        if (s_bellPlayer)
+        // GH#12258: We learned that if you leave the MediaPlayer open, and
+        // press the media keys (like play/pause), then the OS will _replay the
+        // bell_. So we have to re-create the MediaPlayer each time we want to
+        // play the bell, to make sure a subsequent play doesn't come through
+        // and reactivate the old one.
```
Comment (miniksa) [inline src/cascadia/TerminalApp/Pane.cpp:1042]: Isn't this kinda BODGY and we should upstream the request for this? AFAIK MediaPlayer is **THE** way to be playing sounds/media now and not some legacy API so it should offer a way of not having it get captured by the media overlay controls.

File: src/cascadia/TerminalApp/Pane.cpp [hunk 2]
```diff
@@ -1061,17 +1030,48 @@ void Pane::_ControlConnectionStateChangedHandler(const winrt::Windows::Foundatio
...
+            std::weak_ptr<Pane> weakThis2{ shared_from_this() };
+            _mediaEndedRevoker = _bellPlayer.MediaEnded(winrt::auto_revoke, [weakThis2](auto&&, auto&&) {
+                if (auto self{ weakThis2.lock() })
+                {
+                    if (self->_bellPlayer)
+                    {
+                        // We need to make sure clear out the current track
+                        // that's being played, again, so that the system can't
+                        // come through and replay it. In testing, we needed to
+                        // do this, closing the MediaPlayer alone wasn't good
```
Comment (miniksa) [inline src/cascadia/TerminalApp/Pane.cpp:1065]: This is seriously BODGY too.

**Review by miniksa — APPROVED** [review]

File: src/cascadia/TerminalApp/Pane.cpp [hunk 3]
```diff
@@ -1061,17 +1030,48 @@ void Pane::_ControlConnectionStateChangedHandler(const winrt::Windows::Foundatio
 
 winrt::fire_and_forget Pane::_playBellSound(winrt::Windows::Foundation::Uri uri)
 {
-    auto weakThis{ shared_from_this() };
+    std::weak_ptr<Pane> weakThis{ shared_from_this() };
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:1033]: FYI there is a `weak_from_this`

File: src/cascadia/TerminalApp/Pane.cpp [hunk 4]
```diff
@@ -1061,17 +1030,48 @@ void Pane::_ControlConnectionStateChangedHandler(const winrt::Windows::Foundatio
...
             const auto source{ winrt::Windows::Media::Core::MediaSource::CreateFromUri(uri) };
             const auto item{ winrt::Windows::Media::Playback::MediaPlaybackItem(source) };
-            s_bellPlayer.Source(item);
-            s_bellPlayer.Play();
+            _bellPlayer.Source(item);
+            _bellPlayer.Play();
+
+            // This lambda will clean up the bell player when we're done with it.
+            std::weak_ptr<Pane> weakThis2{ shared_from_this() };
+            _mediaEndedRevoker = _bellPlayer.MediaEnded(winrt::auto_revoke, [weakThis2](auto&&, auto&&) {
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:1057]: what series of callbacks do we get for multiple bells?

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalApp/Pane.cpp (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp:1033]: gods I coulda swore that didn't compile

File: src/cascadia/TerminalApp/Pane.cpp (hunk 4 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp:1057]: If you bell while another is playing (in this pane), we'll stop the first bell and replace it with the second one, and only get MediaEnded when the last bell in that pane comes to an end.

**Pull request merged**